### PR TITLE
Replace OIDC backend and custom pipeline step with those from edx-auth-backends

### PR DESCRIPTION
**Pull request by rlucioni**
This replaces the dashboard's OIDC backend and custom pipeline step with those from the new [edx-auth-backends](https://github.com/edx/auth-backends) package.

@clintonb, could you review this when you get a chance?

### CODE REVIEW

File: analytics_dashboard/core/tests/test_backends.py [hunk 1]
```diff
@@ -1,38 +0,0 @@
...
-        claim_name = settings.COURSE_PERMISSIONS_CLAIMS[0]
-        data[claim_name] = [DUMMY_AUTHORIZED_COURSE]
-
-        return data
-
-    def test_login(self):
-        user = self.do_login()
-        self.assertIsNotNone(user)
-
-    def test_course_permissions(self):
```
Comment (clintonb) [inline analytics_dashboard/core/tests/test_backends.py:33]: This is still needed, as this is Insights-specific.

**Comment by clintonb** [comment]
And...I merged a PR that causes a conflict. Sorry about that.

File: analytics_dashboard/core/tests/test_backends.py (hunk 1 above)
Comment (rlucioni) [inline analytics_dashboard/core/tests/test_backends.py:33]: This should be good to go.

**Comment by clintonb** [comment]
:+1:

**Pull request merged**